File: 011.ContainerWithMostWater/wyrj/ContainerWithMostWater.js
/* 132ms */
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    var i = 0, j = height.length - 1;
    var ret = 0;
    while(i < j) {
        var c = Math.min(height[i], height[j]) * (j - i);
        if (c > ret) {
            ret = c;
        }
        (height[i] > height[j]) ? j-- : i++;
    }
    return ret;
};

File: 011.ContainerWithMostWater/wyrj/ContainerWithMostWater.c
/* 8ms */
int maxArea(int* height, int heightSize) {
    int i = 0, j = heightSize - 1;
    int ret = 0, c;
    while(i < j) {
        if (height[i] > height[j]) {
            c = height[j] * (j-i);
            j--;
        } else {
            c = height[i] * (j-i);
            i++;
        }
        if (c > ret) {
            ret = c;
        }
    }
    return ret;
}